src/distances.jl

```diff
@@ -24,53 +32,76 @@ julia> Ellipsoidal([1.0,0.5], [π/2])
 julia> Ellipsoidal([1.0,0.5,0.5], [π/2,0.0,0.0])
 ```
 """
+
+# Right-hand rule; motion defined looking to the negative direction of the axis
+struct RotationRules
+  rot_seq::Symbol
+  motion::Vector{Symbol}
+  radian::Bool
+  main::Symbol
+  extrinsic::Bool
+end
+
+themes = Dict(
+  :TaitBryanExtr => RotationRules(:ZXY,[:CCW,:CCW,:CCW],true,:x,true),
+  :TaitBryanIntr => RotationRules(:ZXY,[:CCW,:CCW,:CCW],true,:x,false),
+  :EulerExtr => RotationRules(:ZXZ,[:CCW,:CCW,:CCW],true,:x,true),
+  :EulerIntr => RotationRules(:ZXZ,[:CCW,:CCW,:CCW],true,:x,false),
+  :GSLIB => RotationRules(:ZXY,[:CW,:CCW,:CCW],false,:y,false),
+  :Leapfrog => RotationRules(:ZYZ,[:CW,:CW,:CW],false,:x,false),
+  :Datamine => RotationRules(:ZXZ,[:CW,:CW,:CW],false,:x,false)
+)
+
 struct Ellipsoidal{N,T} <: Metric
   dist::Mahalanobis{T}
 
-  function Ellipsoidal{N,T}(semiaxes, angles) where {N,T}
+  function Ellipsoidal{N,T}(semiaxes, angles; convention=:TaitBryanExtr) where {N,T}
     @assert length(semiaxes) == N "number of semiaxes must match spatial dimension"
     @assert all(semiaxes .> zero(T)) "semiaxes must be positive"
     @assert N ∈ [2,3] "dimension must be either 2 or 3"
 
+    # invert x and y if necessary
+    if themes[convention].main == :y
```

Comment:
I would first create a rule object with `rule = rules[convention]`, possibly checking first if the provided convention is a key of the dictionary and throwing an error otherwise. Then, we could use the rule object in the rest of the code like `rule.main`, `rule.sequence`, ...

Reply:
Nice. Done it